fondant/component_spec.py

```diff
@@ -196,101 +103,41 @@ def _validate_spec(self) -> None:
         validator = Draft4Validator(spec_schema, resolver=resolver)
 
         try:
-            validator.validate(self.yaml_spec)
+            validator.validate(self._specification)
         except jsonschema.exceptions.ValidationError as e:
             raise InvalidComponentSpec.create_from(e)
 
-    def write_kubeflow_component(self, path: str):
-        """
-        Function that writes the component yaml file required to compile a Kubeflow pipeline
-        """
-        with open(path, "w") as file:
-            yaml.dump(
-                self.kubeflow_component_specification,
-                file,
-                indent=4,
-                default_flow_style=False,
-                sort_keys=False,
-            )
-
-    @property
-    def kubeflow_component_specification(self) -> t.Dict[str, any]:
-        """
-        The Kubeflow component specification
-        """
-        inputs = self.default_input_arguments
-        outputs = self.default_output_arguments
-
-        for arg_name, arg_info in self.yaml_spec["args"].items():
-            inputs.append(
-                KubeflowInput(
-                    name=arg_name.strip(),
-                    description=arg_info["description"].strip(),
-                    type=python2kubeflow_type[arg_info["type"].strip()],
-                )
-            )
+    @classmethod
+    def from_file(cls, path: str) -> "ComponentSpec":
+        """Load the manifest from the file specified by the provided path"""
```

Comment:
Do you mean loading the component spec instead of the manifest here? As the manifest doesn't contain thiings like `input_subsets` and `output_subsets`